source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers chown slurm: /etc/slurm/slurm.conf
+
+   #. Start the slurm service and set it automatically start on boot.
+
+      .. code-block:: bash
+         
+	 sudo pdsh -P workers systemctl enable slurmd --now
+
+   #. Verify the slurmd service is running.
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers systemctl status slurmd | grep Active
+      
+      Example output:
+
+      .. code-block:: console
+
+         hpc-worker3:      Active: active (running) since Wed 2020-04-15 19:39:22 UTC; 1min 17s ago
+         hpc-worker4:      Active: active (running) since Wed 2020-04-15 19:39:22 UTC; 1min 17s ago
+         hpc-worker2:      Active: active (running) since Wed 2020-04-15 19:39:23 UTC; 1min 17s ago
+         hpc-worker1:      Active: active (running) since Wed 2020-04-15 19:39:23 UTC; 1min 17s ago
+
+#. On the controller node, copy the munge key to all nodes and start the 
+   munge service.
+          
+   a. Create the :file:`/etc/munge/` directory on each node.
+
+      .. code-block:: bash
+         
+	 sudo pdsh -P workers mkdir /etc/munge
+   
+   #. Copy the munge key over.
+
+      .. code-block:: bash
+         
+	 sudo pdcp -P workers /etc/munge/munge.key /etc/munge
+
+   #. Set the ownership of the :file:`munge.key` file to `munge`.
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers chown munge: /etc/munge/munge.key
+
+    #. Start the munge service.
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers systemctl enable munge --now
+
+    #. Verify the munge service is running.
+
+       .. code-block:: bash
+
+          sudo pdsh -P workers "systemctl status munge | grep Active"
+      
+       Example output:
+
+       .. code-block:: console
+
+          hpc-worker3:      Active: active (running) since Wed 2020-04-15 19:47:58 UTC; 55s ago
+          hpc-worker4:      Active: active (running) since Wed 2020-04-15 19:47:58 UTC; 55s ago
+          hpc-worker2:      Active: active (running) since Wed 2020-04-15 19:47:59 UTC; 54s ago
+          hpc-worker1:      Active: active (running) since Wed 2020-04-15 19:47:59 UTC; 54s ago
+
+Verify controller can run jobs on all nodes
+*******************************************
+
+#. Check the state of the worker nodes.  
+
+   .. code-block:: bash
+
+      sinfo
+
+   Example output:
+
+   .. code-block:: console
+
+      PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
+      workers*     up   infinite      4   idle hpc-worker[1-4]
+      debug        up   infinite      2   idle hpc-worker[1,3]
+
+   .. tip::
+
+      If the nodes are in a "down" state, put them in the "idle" state.
+
+      .. code-block:: bash
+
+         sudo scontrol update nodename=<worker> state=idle reason=""
+
+#. And finally, verify slurm can run jobs on all 4 worker nodes by issuing
+   a simple :command:`hostname` command.
+
+   .. code-block:: bash
+
+      srun -N4 -p worker hostname
+   
```

Comment:
-p workers, missing 's' from the partition name.

Reply:
Great catch!